app/services/file_creator.rb

```diff
@@ -0,0 +1,34 @@
+class FileCreator
+  attr_reader :data, :format
+
+  def initialize(data, format: :csv)
+    @data = data
+    @format = format
+  end
+
+  def file
+    @file ||= begin
+      file = Tempfile.new 'forams', path
+      file.write data.send("to_#{format}")
+      file.close
+      file
+    end
+  end
+
+  def content_type
+    case format
+    when :csv then 'text/csv'
+    else 'text/plain'
+    end
+  end
+
+  private
+
+  def path
+    @path ||= begin
+      path = File.join('/tmp', 'foram_api', 'forams')
+      FileUtils.mkdir_p path unless File.exists? path
```

Comment:
Tracimy po prostu do niej dostęp. Żadna strata, ponieważ by to była straszna nieskończona rekurencja w tym miejscu